src/utils/Subscription.js

```diff
@@ -1,45 +1,61 @@
 // encapsulates the subscription logic for connecting a component to the redux store, as
 // well as nesting subscriptions of descendant components, so that we can ensure the
 // ancestor components re-render before descendants
+
+function initListeners() {
+  let count = 0
+  let current = []
+  let next = []
+
+  return {
+    clear() {
+      count = 0
+      next = null
```

Comment:
See prev comment about count. Setting the arrays to null means if somehow code gets back in here and tries to operator on the null arrays, it will throw an error. `subscribe` should never get called again once it's been unsubscribed, so creating empty arrays would just be extra object allocation.